src/tp_engine.hpp

```diff
@@ -141,9 +216,16 @@ class TPEngine {
   size_t edge_count_ = 0;
   size_t edge_alloc_ = 0;
   size_t edge_spare_count_ = 3;
+  // Total number of trees used to construct the DAG.
+  size_t tree_count_ = 0;
```

Comment:
Its purpose was just so when I set which tree each edge came from (the best tree containing the given edge), I needed the max count for unassigned edges.  It could change it to `input_tree_count_`?